release.sh

```diff
@@ -0,0 +1,203 @@
+    read -r -e -p "Enter a target release (i.e: v0.1.2): " RELEASE_VERSION
+    [[ -z ${RELEASE_VERSION} ]] && {
+      echo "no target release"
+      exit 1
+    }
+  }
+  [[ ${RELEASE_VERSION} =~ v[0-9]+\.[0-9]*\.[0-9]+ ]] || {
+    echo "invalid version provided, need to match v\d+\.\d+\.\d+"
+    exit 1
+  }
+}
+
+buildDbMigrationImage() {
+  info Building DB Migration Image
+  echo -----------------------------------
+  cd "$API_DIR"
+  docker build -f db.Dockerfile -t quay.io/tekton-hub/db-migration:${RELEASE_VERSION} . && docker push quay.io/tekton-hub/db-migration:${RELEASE_VERSION}
+  info DB Migration Image Build Successfully
+  echo -----------------------------------
+}
+
+buildApiImage() {
+  info Building API Image
+  echo -----------------------------------
+  cd "$API_DIR"
+  docker build -t quay.io/tekton-hub/api:${RELEASE_VERSION} . && docker push quay.io/tekton-hub/api:${RELEASE_VERSION}
+  info API Image Build Successfully
+  echo -----------------------------------
+}
+
+buildUiImage() {
+  info Building UI Image
+  echo -----------------------------------
+  cd "$UI_DIR"
+  docker build -t quay.io/tekton-hub/ui:${RELEASE_VERSION} . && docker push quay.io/tekton-hub/ui:${RELEASE_VERSION}
+  info UI Image Build Successfully
+  echo -----------------------------------
+}
+
+db(){
+	info Creating DB Release Yaml
+
+  make db || {
+    err 'db release build failed'
+    return 1
+  }
+}
+
+db-migration(){
+	info Creating Db-Migration Release Yaml
+
+  make db-migration || {
+    err 'db-migration release build failed'
+    return 1
+  }
+}
+
+api-k8s(){
+	info Creating API Release Yaml
+
+  make api-k8s || {
+    err 'api release build failed'
+    return 1
+  }
+}
+
+api-openshift(){
+	info Creating API Release Yaml
+
+  make api-openshift || {
+    err 'api release build failed'
+    return 1
+  }
+}
+
+ui-k8s(){
+	info Creating UI Release Yaml
+
+  make ui-k8s || {
+    err 'ui release build failed'
+    return 1
+  }
+}
+
+ui-openshift(){
+	info Creating UI Release Yaml
+
+  make ui-openshift || {
+    err 'ui release build failed'
+    return 1
+  }
+}
+
+replaceImageName() {
+  info Changing Image Name
+
+  cd "$RELEASES_DIR"
+
+  #  Replace the db-migration image name
+  sed -i "s@image: quay.io/tekton-hub/db-migration@image: quay.io/tekton-hub/db-migration:$RELEASE_VERSION@g" db-migration.yaml
+
+  # Replace the api image
+  sed -i "s@image: quay.io/tekton-hub/api@image: quay.io/tekton-hub/api:$RELEASE_VERSION@g" api-k8s.yaml
+
+  sed -i "s@image: quay.io/tekton-hub/api@image: quay.io/tekton-hub/api:$RELEASE_VERSION@g" api-openshift.yaml
+
+  #Replace the ui image
+  sed -i "s@image: quay.io/tekton-hub/ui@image: quay.io/tekton-hub/ui:$RELEASE_VERSION@g" ui-k8s.yaml
+
+  sed -i "s@image: quay.io/tekton-hub/ui@image: quay.io/tekton-hub/ui:$RELEASE_VERSION@g" ui-openshift.yaml
+}
+
+createGitTag() {
+  echo; echo 'Creating tag for new release:'
+
+  hub release create -a $RELEASES_DIR/db-migration.yaml \
```

Comment:
will it be final release or pre-release, or we need to do something specifically after this ?

Reply:
It will be a pre-release, after adding the release notes we can publish it